### PR TITLE
refactor(checkbox-group): refactor checkbox group to use fieldset - FE-3412 

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Current behaviour
Checkbox-group doesn't use Fieldset

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] <del> Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: cypress/features/regression/accessibility/accessibilityExperimental.feature [hunk 1]
```diff
@@ -69,13 +69,12 @@ Feature: Accessibility tests - Experimental folder
...
-    When I open "Experimental checkbox" component page "<story>" in no iframe
+    When I open "Experimental checkbox validations" component page "<story>" in no iframe
     Then "Experimental checkbox <story> story" component has no accessibility violations
     Examples:
       | story                              |
-      # Commented because of issue FE-3412
-       #| group checkbox boolean validation  |
-       #| group checkbox string validation   |
+      | group checkbox boolean validation  |
+      | group checkbox string validation   |
```
Comment (Alec-Walker-Sage) [inline cypress/features/regression/accessibility/accessibilityExperimental.feature:77]: Cypress tests run and pass but when I visit these stories on chrome it shows that there is an accessibility issue 
image

Comment (DlgSHi) [reply cypress/features/regression/accessibility/accessibilityExperimental.feature:77]: it is because we've disabled `WGAC 2.0 a` and `WGAC 2.0 aa`

Comment (Alec-Walker-Sage) [reply cypress/features/regression/accessibility/accessibilityExperimental.feature:77]: ah had forgotten about this

**Review by Alec-Walker-Sage — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

File: src/__experimental__/components/checkbox/checkbox-group.component.js [hunk 2]
```diff
@@ -2,42 +2,73 @@ import React from "react";
...
+    legendAlign,
+    legendSpacing,
+  } = props;
 
   const groupLabelId = `${groupName}-label`;
 
   return (
-    <InputGroupBehaviour>
+    <Fieldset
+      role="group"
```
Comment (dawidzarzycki) [inline src/__experimental__/components/checkbox/checkbox-group.component.js:27]: After I took a look at this https://www.w3.org/WAI/tutorials/forms/grouping/ I think we don't need to use `role="group"` on this `Fieldset` element

File: src/__experimental__/components/checkbox/checkbox-group.component.js [hunk 3]
```diff
@@ -2,42 +2,73 @@ import React from "react";
...
+      legendSpacing={legendSpacing}
+      error={error}
+      warning={warning}
+      info={info}
+      mb={mb}
+      isRequired={required}
+      {...tagComponent("checkboxgroup", props)}
+    >
       <StyledCheckboxGroup
         aria-labelledby={groupLabelId}
```
Comment (dawidzarzycki) [inline src/__experimental__/components/checkbox/checkbox-group.component.js:41]: I see that you didn't change the line but I can't find related `ID` to this `aria-labelledby` so probably it doesn't work as it should

File: src/__experimental__/components/checkbox/checkbox-group.component.js (hunk 2 above)
Comment (grabkowski) [reply src/__experimental__/components/checkbox/checkbox-group.component.js:27]: actually we don't need to use any role on this element and it's descendant, I'll remove both of them

File: src/__experimental__/components/checkbox/checkbox-group.component.js (hunk 3 above)
Comment (grabkowski) [reply src/__experimental__/components/checkbox/checkbox-group.component.js]: removed as it's already labelled in a fieldset

**Review by sjohnson-sage — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**